### PR TITLE
Let filter=todo not return results with statuses Claimed,TempFail

### PR DESCRIPTION
**Pull request by lablans**

### CODE REVIEW

**Comment by lablans** [comment]
@TKussel , we might run into diverging code if we leave this open for too long.

**Comment by TKussel** [comment]
With the tests working -- even if I don't understand why -- I'm for merging this branch.

**Comment by TKussel** [comment]
If I understand correctly, the changes in PR #53 render the `std::mem::discriminant` trickery obsolete?

**Comment by lablans** [comment]
@TKussel please check and merge if this is ready. This PR contains important changes to the Readme.

File: README.md [hunk 1]
```diff
@@ -223,7 +223,7 @@ Parameters:
 - `filter` (optional): Fetch only tasks fulfilling the specified filter criterion. Generic queries are not yet implemented, but the following "convenience filters" reflecting common use cases exist:
   - `filter=todo`: Matches unfinished tasks to be worked on by the asking client. Is a combination of:
     - `to` contains me and
-    - `results` do not contain me.
+    - `results` do not contain a result from me (results with `status` in `claimed,tempfail` are ignored).
```
Comment (TKussel) [inline README.md:226]:
```suggestion
    - `results` do not contain a result from me (except results with `status` values of `claimed,tempfail`, to allow resuming those tasks).
```

**Review by TKussel — APPROVED** [review]
I find the Readme sentence ambiguous, hence, I suggested a change.
I fixed some errors in the unit tests, they now work as intended.

**Pull request merged**